server/mqtt.go

```diff
@@ -4711,29 +4723,28 @@ func (sess *mqttSession) processJSConsumer(c *client, subject, sid string,
 			ccr.Config.InactiveThreshold = opts.MQTT.ConsumerInactiveThreshold
 		}
 		if _, err := sess.jsa.createConsumer(ccr); err != nil {
+			// <>/<> FIXME handle already exists error
 			c.Errorf("Unable to add JetStream consumer for subscription on %q: err=%v", subject, err)
 			return nil, nil, err
 		}
 		cc = &ccr.Config
-		sess.tmaxack += maxAckPending
+
+		tmaxack += maxAckPending
 	}
+
 	// This is an internal subscription on subject like "$MQTT.sub.<nuid>" that is setup
 	// for the JS durable's deliver subject.
 	sess.mu.Lock()
-	sub, err := c.processSub([]byte(inbox), nil, []byte(inbox), mqttDeliverMsgCbQoS12, false)
+	sess.tmaxack = tmaxack
+	sub, err := sess.processSub(c, []byte(inbox), []byte(inbox),
```

Comment:
I have a problem with that in that the original code dealt only with `sub` where calling this now makes it process the shadow subs. Unless we understand that this was a bug, for the sake of minimizing behavior changes, I would not do that here.

Reply:
ok

Reply:
See https://github.com/nats-io/nats-server/pull/4760/commits/a5e47c99aaa7ec2b01a5e7bb4fe0daa3b765a422